12_TriggerImp.sql

```diff
@@ -0,0 +1,37 @@
+IF NOT EXISTS (
+    SELECT * FROM sys.tables t 
+    JOIN sys.schemas s ON (t.schema_id = s.schema_id) 
+    WHERE s.name = 'dbo' AND t.name = 'AuditLog') 
+    CREATE TABLE AuditLog (
+        LogID INT PRIMARY KEY IDENTITY(1,1),
+        BookID INT,
+        StatusChangeTo VARCHAR(50),
+		ChangeDate DATETIME
+    );
+GO
+CREATE OR ALTER TRIGGER trg_LogBooksStatusChanges
+ON Books
+AFTER UPDATE 
+AS
+BEGIN
+    INSERT INTO AuditLog (BookID, StatusChangeTo, ChangeDate)
+    SELECT 
+        BookID,
+        CASE
+			WHEN  CurrentStatus  = 'Available' THEN 'Available'
+		    WHEN  CurrentStatus  = 'Borrowed' THEN 'Borrowed'
+
+        END AS StatusChange,
+        GETDATE() AS ChangeDate
+    FROM inserted
+    WHERE EXISTS (SELECT 1 FROM deleted WHERE deleted.BookID = inserted.BookID AND deleted.CurrentStatus != inserted.CurrentStatus);
```

Comment:
Can't we just join to deleted table her ?
join might be optimized, (not sure how it perform here) but if there any index it would utilize it.